src/components/feedback/Dialog.tsx

```diff
@@ -48,46 +49,77 @@ type ComponentProps = {
    * Optional TransitionComponent for open (mount) and close (unmount) transitions
    */
   transitionComponent?: TransitionComponent;
-
-  /**
-   * Dialog lays out its content in a Panel by default, but `custom` allows any
-   * layout of Dialog children
-   */
-  variant?: 'panel' | 'custom';
 };
 
 // This component forwards a number of props on to `Panel` but always sets the
 // `fullWidthHeader` prop to `true`.
-export type DialogProps = PresentationalProps &
+export type PanelDialogProps = PresentationalProps &
   ComponentProps &
-  Omit<PanelProps, 'fullWidthHeader'>;
+  Omit<PanelProps, 'fullWidthHeader'> & {
+    /** Dialog lays out its content in a Panel by default */
+    variant?: 'panel';
+  };
+
+type NonPanelDialogProps = PresentationalProps &
+  ComponentProps & {
+    /** `custom` allows any layout of Dialog children */
+    variant: 'custom';
+
+    /** @deprecated Title is ignored in custom dialogs */
+    title?: string;
```

Comment:
I've added this only for backwards compatibility, otherwise any consumer which was using `variant="custom"` will now get an error, as the title was mandatory up until now. But I feel tempted to just remove it.

Reply:
I just noticed `title` gets marked as deprecated even when using `variant="panel"'`, so this might not be the best idea in the end.

Reply:
> But I feel tempted to just remove it.

I would just do this. If we were being strict about semver we'd just bump the major version.

Reply:
I'll remove it then. At the end of the day, we are most probably the only ones using this library.